src/DataCollection.Shared/ViewModels/AuthViewModel.cs

```diff
@@ -55,6 +55,15 @@ public AuthViewModel(string webmapURL, string arcGISOnlineURL, string appClientI
             _userName = userName;
             _oAuthRefreshToken = oAuthRefreshToken;
 
+            // Watch for oAuth token changes
+            BroadcastMessenger.Instance.BroadcastMessengerValueChanged += (o, args) =>
```

Comment:
This appears to only update `_oAuthRefreshToken` when it's been cleared out.  That seems a bit peculiar.  If we need to handle token updates, I'd expect that the handler would respond to any update.